<!--#######################################-->
<!-- Copyright (c) 2024  Dong Zhaorui.     -->
<!-- All rights reserved.                  -->
<!-- Author: Dong Zhaorui 847235539@qq.com -->
<!-- Date  : 2024-12-23                    -->
<!--#######################################-->

<launch>
    # arg
    <arg name="test_flag" default="true" />
    <arg name="visual_flag" default="true" />
    <arg name="sim_flag" default="true" />
    # use sim time
    <param name="use_sim_time" value="$(arg sim_flag)" />

    # mpc track
    <node name="mpc_track" pkg="hex_toolkit_echo_plus" type="mpc_track.py" output="screen" >
        # param
        <param name="model_path" value="$(find hex_toolkit_echo_plus)/urdf/echo_plus.urdf" />
        <param name="mpc_root" value="$(find hex_toolkit_echo_plus)" />
        <rosparam file="$(find hex_toolkit_echo_plus)/config/ros1/mpc_track.yaml"/>
        # subscribe
        <remap from="/chassis_odom" to="/odom" />
        <remap from="/target_pose" to="/target_pose" />
        # publish
        <remap from="/unsafe_ctrl" to="/cmd_vel" />
        <remap from="/vel_ctrl" to="/unused" />
    </node>

    # test
    <group if="$(arg test_flag)">
        # cartesian target generator
        <node name="cart_gen" pkg="hex_toolkit_general_chasssis" type="cart_gen.py" output="screen" >
            # param
            <rosparam file="$(find hex_toolkit_general_chasssis)/config/ros1/cart_gen.yaml"/>
            # publish
            <remap from="/target_pose" to="/target_pose" />
        </node>
    </group>

    # bringup
    <include file="$(find hex_toolkit_echo_plus)/launch/ros1/bringup.launch">
        <arg name="visual_flag" value="$(arg visual_flag)" />
        <arg name="sim_flag" value="$(arg sim_flag)" />
    </include>
</launch>
